bundles/admin/admin-hierarchical-layerlist/components/Layer.js

```diff
@@ -0,0 +1,246 @@
+                groups.push({
+                    id: subgroup.id,
+                    cls: 'subgroup',
+                    name: Oskari.getLocalized(subgroup.name)
+                });
+
+                // subgroup subgroups
+                subgroup.groups.forEach(function(subgroupsubgroup) {
+                    groups.push({
+                        id: subgroupsubgroup.id,
+                        cls: 'subgroupsubgroup',
+                        name: Oskari.getLocalized(subgroupsubgroup.name)
+                    });
+                });
+            });
+        });
+
+        return groups;
+    },
+
+    /*******************************************************************************************************************************
+    /* PUBLIC METHODS
+    *******************************************************************************************************************************/
+    /**
+     * Get dataproviders
+     * @method getDataproviders
+     * @param  {Boolean}        emptyCache empty cache
+     * @param  {Function}       callback   callback function
+     */
+    getDataproviders: function(emptyCache, callback) {
+        var me = this;
+        if (me.dataProviders === null || emptyCache === true) {
+            jQuery.ajax({
+                type: 'GET',
+                dataType: 'json',
+                contentType: 'application/json; charset=UTF-8',
+                url: me.sandbox.getAjaxUrl('GetMapLayerGroups'),
+                error: function() {
+                    var errorDialog = Oskari.clazz.create('Oskari.userinterface.component.Popup');
+                    errorDialog.show(me.locale('errors.dataProvider.title'), me.locale('errors.dataProvider.message'));
+                    errorDialog.fadeout();
+                },
+                success: function(response) {
+                    me.dataProviders = [];
+                    response.organization.forEach(function(org) {
+                        me.dataProviders.push({
+                            id: org.id,
+                            name: Oskari.getLocalized(org.name)
+                        });
+                    });
+
+                    me.dataProviders.sort(function(a, b) {
+                        return Oskari.util.naturalSort(a.name, b.name);
+                    });
+                    callback();
+
+                }
+            });
+        }
+        callback();
+    },
+    /**
+     * Shows layer popup
+     * @method showLayerAddPopup
+     * @param  {String}          tool    tool identifier
+     * @param  {String}          layerId layerId
+     * @param  {String}          groupId groupId
+     */
+    showLayerAddPopup: function(tool, layerId, groupId) {
+        var me = this;
+        me.getDataproviders(false, function() {
+            me._extraFlyout.move(100, 100, true);
+            me._extraFlyout.makeDraggable({
+                handle: '.oskari-flyouttoolbar, .statsgrid-data-container > .header',
```

Comment:
Why would this handle reference statsgrid? I'm not sure you need to send any opts as param for this.